lib/ContractInteract/GatewayComposer.js

```diff
@@ -0,0 +1,445 @@
+'use strict';
+
+const Web3 = require('web3');
+const Contracts = require('../Contracts');
+const AbiBinProvider = require('../AbiBinProvider');
+const Utils = require('../../utils/Utils');
+
+const { BN } = Web3.utils;
+
+const ContractName = 'GatewayComposer';
+
+/**
+ * Contract interact for Gateway composer.
+ */
+class GatewayComposer {
+  /**
+   * Constructor for Gateway composer.
+   *
+   * @param {Web3} web3 Web3 instance that points to origin.
+   * @param {string} address Gateway composer contract address.
+   */
+  constructor(web3, address) {
```

Comment:
Should we rename web3 to originWeb3 as GatewayComposer is deployed on origin chain?